components/service/glean/src/test/java/mozilla/components/service/glean/GleanTest.kt

```diff
@@ -263,8 +263,6 @@ class GleanTest {
             Glean.handleBackgroundEvent()
         }
 
-        Glean.pingStorageEngine.testWait()
```

Comment:
I think this would still be needed to prevent intermittent tests, no?

Reply:
Nope, since the ping storage now runs on a Dispatcher.API.launch thread inside of sendPing(), we await it just like we do all the other background threads

Reply:
As in, we don't have to do anything special to await it anymore, we can treat it like the other storage engines.

Reply:
Don't writes happen on Kotlin's [`Dispatcher.IO`](https://github.com/mozilla-mobile/android-components/blob/5de8b4c655db8dfbc19e344c7ad2a854e1047818/components/service/glean/src/main/java/mozilla/components/service/glean/storages/PingStorageEngine.kt#L61)?

Reply:
Individually, yes, but all of those tasks are launched from inside of `sendPings()` which launches all of those `.IO` tasks from `Dispatchers.API`

Reply:
This is what allows us to await the writes off of the main thread.

Reply:
To explain more clearly:
1. We need to send some pings so someone calls `Glean.sendPings()`
2. Send pings spawns a `Dispatchers.API` thread, which in turn spawns each of the `Dispatcher.IO` threads that are responsible for writing a single ping to disk.
3. Inside of the `Dispatchers.API` thread we await the `Dispatchers.IO` tasks to be completed before enqueueing the `PingUploadWorker`.  For tests, we await `Dispatchers.API` synchronously, otherwise the awaiting is happening in the background thread so not to block the main glean thread.

Reply:
Right, understood! Thanks for the explanation!